systemtest/profiling_test.go

```diff
@@ -82,31 +83,20 @@ func TestProfiling(t *testing.T) {
 
 	const secretToken = "test_token"
 
-	apmIntegration := newAPMIntegrationConfig(t,
-		map[string]interface{}{
-			"secret_token": secretToken,
-		}, map[string]interface{}{
-			"apm-server": map[string]interface{}{
-				"value": map[string]interface{}{
-					"profiling": map[string]interface{}{
-						"enabled": true,
-						"elasticsearch": map[string]interface{}{
-							"api_key": apiKey.ID + ":" + apiKey.APIKey,
-						},
-						// A separate elasticsearch configuration is required
-						// for host agent metrics. In practice this would likely
-						// be a different cluster, with a different API Key,
-						// but we're not going to those lengths for a system test.
-						"metrics.elasticsearch": map[string]interface{}{
-							"hosts":   []string{"elasticsearch:9200"},
-							"api_key": apiKey.ID + ":" + apiKey.APIKey,
-						},
-					},
-				},
-			},
-		})
-
-	apmServerURL, _ := url.Parse(apmIntegration.URL)
+	srv := apmservertest.NewUnstartedServerTB(t)
```

Comment:
With this change, the comment at line 58 doesn't make sense, and the test doesn't quite cover what it was doing. We could simulate what Elastic Agent does by creating a more limited API Key for the Elasticsearch output.

Reply:
Good catch! Wouldn't the behaviour be the same with standalone ? In this case: the standalone APM Server is using the default elasticsearch credentials (with narrow privileges) and we are creating and passing an api_key to write to the profiling indices.

Reply:
Ah, true. I was thinking the standalone apm-server run by system tests used superuser privileges, but they don't. So, I think we just need to update the comment then.

Reply:
Updated! :+1: